ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3SubsystemAdd.java

```diff
@@ -337,8 +339,8 @@ protected void performBoottime(OperationContext context, ModelNode operation, Re
         this.denyAccessByDefault.set(defaultMissingMethodValue);
 
         final ModelNode defaultStatefulSessionTimeout = EJB3SubsystemRootResourceDefinition.DEFAULT_STATEFUL_BEAN_SESSION_TIMEOUT.resolveModelAttribute(context, model);
-        final ValueService<AtomicLong> defaultStatefulSessionTimeoutService = new ValueService<>(new ImmediateValue<>(
-                defaultStatefulSessionTimeout.isDefined() ? new AtomicLong(defaultStatefulSessionTimeout.asLong()) : DefaultStatefulBeanSessionTimeoutWriteHandler.INITIAL_TIMEOUT_VALUE));
+        final ValueService<AtomicLong> defaultStatefulSessionTimeoutService = new ValueService<>(() ->
+                defaultStatefulSessionTimeout.isDefined() ? new AtomicLong(defaultStatefulSessionTimeout.asLong()) : DefaultStatefulBeanSessionTimeoutWriteHandler.INITIAL_TIMEOUT_VALUE);
```

Comment:
We still want resolve to resolve the default timeout eagerly, that way we can use a Service that provides a fixed value.

Reply:
Fixed